spki/src/traits.rs

```diff
@@ -115,8 +118,15 @@ pub trait DynAssociatedAlgorithmIdentifier {
 }
 
 #[cfg(feature = "alloc")]
-impl<T: AssociatedAlgorithmIdentifier> DynAssociatedAlgorithmIdentifier for T {
+impl<T> DynAssociatedAlgorithmIdentifier for T
+where
+    T: AssociatedAlgorithmIdentifier,
+    T::Params: Into<Any>,
```

Comment:
Debatably we could get rid of this bound by using `Any::encode_msg`, however that's fallible so we'd have to add an `expect` and panic on encoding failure.